compiler/p/codegen/OMRMemoryReference.cpp

```diff
@@ -1743,6 +1749,12 @@ void OMR::Power::MemoryReference::accessStaticItem(TR::Node *node, TR::SymbolRef
          loadAddressConstant(cg, cg->needRelocationsForBodyInfoData(), nodeForSymbol, 0, reg, NULL, false, TR_BodyInfoAddressLoad);
          return;
          }
+      else if (cg->comp()->compileRelocatableCode() && symbol->isCatchBlockCounter())
```

Comment:
It's better to replace `compileRelocatableCode()` with something like `needRelocationsForBodyInfoData()`